tests/schema/tests.py

```diff
@@ -5210,6 +5216,225 @@ def test_alter_field_add_db_index_to_charfield_with_unique(self):
+            self.get_constraints_for_column(AuthorCharFieldWithIndex, "char_field"),
+            [
+                "schema_authorcharfieldwithindex_char_field_06a11776",
+                "schema_authorcharfieldwithindex_char_field_06a11776_like",
+            ],
+        )
+
+    @unittest.skipUnless(connection.vendor == "postgresql", "PostgreSQL specific")
+    def test_charfield_change_max_length_no_index_created(self):
+        with connection.schema_editor() as editor:
+            editor.create_model(Author)
+        self.assertEqual(
+            self.get_constraints_for_column(Author, "name"),
+            [],
+        )
+        old_name = Author._meta.get_field("name")
+        new_name = CharField(max_length=20)
+        new_name.set_attributes_from_name("name")
+        with connection.schema_editor() as editor:
+            editor.alter_field(Author, old_name, new_name, strict=True)
+        self.assertEqual(
+            self.get_constraints_for_column(Author, "name"),
+            [],
+        )
+
+    @isolate_apps("schema")
+    @unittest.skipUnless(connection.vendor == "postgresql", "PostgreSQL specific")
+    def test_slugfields_change_primary_key(self):
+        class SimpleModel(Model):
+            field1 = SlugField(max_length=20, primary_key=True)
+            field2 = SlugField(max_length=20)
+
+            class Meta:
+                app_label = "schema"
+
+        with connection.schema_editor() as editor:
+            editor.create_model(SimpleModel)
+        self.assertEqual(
+            self.get_constraints_for_column(SimpleModel, "field1"),
+            [
+                "schema_simplemodel_field1_f07a3d6a_like",
+                "schema_simplemodel_pkey",
+            ],
+        )
+        # remove primary_key from field1
+        old_field1 = SimpleModel._meta.get_field("field1")
+        new_field1 = CharField(max_length=20, primary_key=False)
+        new_field1.set_attributes_from_name("field1")
+        with connection.schema_editor() as editor:
+            editor.alter_field(SimpleModel, old_field1, new_field1, strict=True)
+        self.assertEqual(self.get_constraints_for_column(SimpleModel, "field1"), [])
+        # add primary_key to field2
+        old_field2 = SimpleModel._meta.get_field("field2")
+        new_field2 = CharField(max_length=20, primary_key=True)
+        new_field2.set_attributes_from_name("field2")
+        new_field2.model = SimpleModel
+        with connection.schema_editor() as editor:
+            editor.alter_field(SimpleModel, old_field2, new_field2, strict=True)
+        self.assertEqual(
+            self.get_constraints_for_column(SimpleModel, "field2"),
+            [
+                "schema_simplemodel_field2_08772539_like",
+                "schema_simplemodel_field2_08772539_pk",
+            ],
+        )
+
+    @unittest.skipUnless(connection.vendor == "postgresql", "PostgreSQL specific")
+    def test_slugfields_change_primary_key_operations(self):
```

Comment:
This test is problematic (if I revert the changes, this errors and then causes ~50 tests to error) 
I believe it's because the table is not cleaned up properly but I also believe it might be already tested through `test_slugfields_change_primary_key` and can be removed

Reply:
I'm not confident that this is fully covered by other tests, but I can remove it if you want.

Reply:
I moved the test into its own class in operations tests, and when I made it fail, it only showed 3 failures for that test class.